app/src/org/commcare/heartbeat/HeartbeatRequester.java

```diff
@@ -156,6 +156,7 @@ private static void parseUpdateToPrompt(JSONObject latestVersionInfo, UpdateToPr
                     if (latestVersionInfo.has("force")) {
                         forceString = latestVersionInfo.getString("force");
                     }
+                    HiddenPreferences.setLatestCommcareVersion(versionValue);
```

Comment:
@shubham1g5 Just noticed one thing. This particular method `parseUpdateToPrompt` is being called from both `attemptCczUpdateParse` and `attemptApkUpdateParse`. 
But for both of them we're using `setLatestCommcareVersion` when we should've used `setLatestAppVersion`. 
I'm not sure about the actual impact of this bug, but seems like we should fix this in this release. Can you please check?

Reply:
ahh good catch. We should wrap this setter in a `updateType` check. The impact would not have been noticeable as no one is using the recovery feature it's used in currently.  Also can we also set this latest version info from the playstore update API for in-app updates ?

Reply:
> can we also set this latest version info from the playstore update API for in-app updates?

Yeah, I think we can get the `versionCode` from the playstore api https://developer.android.com/reference/com/google/android/play/core/appupdate/AppUpdateInfo#availableVersionCode(), but not the version name.

Reply:
ok we do need the version name here to be compatible with the surrounding code.

Reply:
https://github.com/dimagi/commcare-android/pull/2509